### ISSUE TITLE
Angular TypeScirpt

### ISSUE THREAD

**Issue by winalex** [consulo/consulo-javascript#17]
Is it possible to use Angular, Type Script with Consulo. If yes how to setup which pluggins are needed???

**Comment by VISTALL** [comment]
Hi. There no open TS implementation (Angular plugin is opened, but it's depends to closed TS implementation).

Few guys asking me for TS, but it's not easy task.

When - idk. No plans for now.

Thanks

**Comment by winalex** [comment]
;'(

**Comment by Sajito** [comment]
> 
> 
> Hi. There no open TS implementation (Angular plugin is opened, but it's depends to closed TS implementation).
> 
> Few guys asking me for TS, but it's not easy task.
> 
> When - idk. No plans for now.
> 
> Thanks

Hi @VISTALL, isn't it possible to integrate the typescript language service?

**Comment by VISTALL** [comment]
Hello @Sajito.

TS it's not only language features. For example debug for nodejs(or browser), npm and build tools.

In theory i can add base implementation, but i need some list what you use for dev(what platform etc). I need understand at what platform i work.

About LSP. I don't want provide LSP services as primary language system, for few reasons - like different LSP services use different runtime - TS require nodejs, .NET require .NET runtime. From my experience - there no ideal runtime, which always works for any user:)

Maybe later LSP will be good as additional analyzer, but not primary.

Thanks

**Comment by Sajito** [comment]
I work quite a lot with PhpStorm, which - as far as I know - uses a combination of it's own language analyzer and TS LSP.

Personally I'd like to be able to be able to create JavaScript or TypeScript applications with consulo. Like an angular app for the browser. It would already help a lot being able to have autocompletion for JS/TS. Debug and build tool integration is not as important as being able to actually write code.

**Comment by VISTALL** [comment]
> as far as I know - uses a combination of it's own language analyzer and TS LSP.

yeah. But own analyzer is primary.

> Personally I'd like to be able to be able to create JavaScript or TypeScript applications with consulo.

Me too:) But i need understand how you use it?(TS or JS)

**Comment by Sajito** [comment]
My primary use case is TS, to delevop angular applications.

**Comment by VISTALL** [comment]
> My primary use case is TS, to delevop angular applications.

web application ? (with php? because phpstorm - not webstorm)

**Comment by Sajito** [comment]
Yes, correct. Most of the time working with Symfony as PHP backend. Frontend stuff mostly TS, sometimes plain JS. But the good thing with a working TS lexer would be, that it automatically supports JS.

The current PHP extension needs some love too, but that one has its own Repo.

**Comment by VISTALL** [comment]
Hmm.

Yeah - PHP plugin has own repository. It's very old plugin (idea plugin was based on it) - it's stub. I added some few features, but there no requests - that why it's became dusty
